### DISCUSSION TITLE
SQLSTATE[HY000]: General error: Datatype Mismatch with SQLite Database

### DISCUSSION THREAD

**Discussion by anshu8858** [laravel/framework#48521] [Q&A]
### Laravel Version

10

### PHP Version

8.2

### Database Driver & Version

sqlite3

### **In Controller**

```
$biz = new Business;
        $biz->business_name = $request->business_name;
        $biz->business_tagline = $request->business_tagline;
        $biz->address =  addslashes($request->string('address')->trim());  // slashes added befor special characters
        $biz->phone = $request->phone;
        $biz->email = $request->email;
        $biz->web = $request->string('web')->trim();
        $biz->city = ucwords($request->city);
        $biz->zipcode = $request->zipcode;
        $biz->country = ucwords($request->country);
        $biz->country_code = $request->country_code;
        $biz->gstin = $request->gstin;
        $biz->contact_person_name = ucwords($request->contact_person_name);
        $biz->is_active = $request->has('is_active')?1:0;
        $biz->user_id = Auth::id();

        if ($request->hasFile('logo')) {

            $photo = $request->file('logo');
            $ext = $request->logo->extension();  // File Extension

            $photoname = Str::uuid(); //Unique ID
            $photoname = $photoname.".".$ext;
            
            $dtpath=$this->filepath.$photoname; //File Logo Path

            $ii = Image::make($photo->getRealPath());
            $ii->fit(360, 360, function ($constraint) {
                $constraint->upsize();
            })->save(public_path('storage/'.$dtpath));
            
            $biz->logo=$dtpath;
            $biz->save();
            $biz->ddRawSql();
            dd(DB::getQueryLog());
        }
        else
        {
            $biz->save();
            dd(DB::getQueryLog());
        }
```

**### SQL Query**

`insert into "businesses" ("business_name", "business_tagline", "address", "phone", "email", "web", "city", "zipcode", "country", "country_code", "gstin", "contact_person_name", "is_active", "user_id", "logo", "id", "uuid", "updated_at", "created_at") values (Business Name, Business Softech, Aati Apartment, Building, A-1,, +91910XXXXX50, abxxxxfd@gmail.com, https://pk.com/u/biz/add, abc, 123456, India, 91, ?, , 1, 3, biz/logo/cc2253e5-a781-426c-b148-72ff9147ff58.jpg, d941213d-4fb7-437e-a2b0-d0de774e7816, 3a146e29-253d-4ef1-b727-6a7a9f7362a6, 2023-09-23 22:06:20, 2023-09-23 22:06:20)`

### Steps To Reproduce

Anybody please give me any solution. I am not understanding that where is the issue. 
I think **Query Builder** generating wrong SQL query for **SQLite**.
In the address field there are extra comma ','. I think all value field should be in **quote**.

**Comment by crynobone** [comment]
Hi there,

Thanks for reporting the problem you are encountering, but it looks like this is a question which may be better suited for a support channel. We only use this issue tracker for reporting bugs with the library itself. If you have a question on how to use functionality provided by this repository you can try one of the following channels:

* [Laracasts Forums](https://laracasts.com/discuss)
* [Laravel.io Forums](https://laravel.io/forum)
* [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
* [Discord](https://discordapp.com/invite/KxwQuKb)
* [Larachat](https://larachat.co/)
* [IRC](https://web.libera.chat/?nick=laravelnewbie&channels=#laravel)

However, this issue will not be locked and everyone is still free to discuss solutions to your problem!

Thanks.
